### PR TITLE
feat: Suggested follows

### PR DESCRIPTION
**Pull request by vinnyjth**
### What does this PR do, or why is it needed?

Adds suggested followers query :D 

Adds a list of people that you should follow, assuming you've added the appropriate config.yml field.
Companion PR is here: https://github.com/ApollosProject/apollos-templates/pull/277

The list is based on user emails for now. We might have to adapt that in the future based on the needs of different clients, but this is probably the best place to start. 

You can also optionally supply `campusIds` when specifying users, and that will cause those users to only show up to users that share a campus with that user. 

### How do I test this PR?

Run this query in templates 

```
query {
  suggestedFollows {
    id
    firstName
    lastName
  }
}
```

### CODE REVIEW

**Comment by vinnyjth** [comment]
Blocked until the following/followers PR is merged

File: packages/apollos-data-connector-postgres/src/people/dataSource.js [hunk 1]
```diff
@@ -83,20 +72,33 @@ export default class Person extends PostgresDataSource {
...
+
+    await this.model.update({ profileImageUrl: url }, { where });
+
+    return this.model.findOne({ where });
+  };
+
+  // Returns a where clause that will find the current person
+  // Raise an error if the current person doesn't exist
+  // Smart enough to work even if the Auth dataSource pulls from a different database than postgres
+  whereCurrentPerson = async () => {
```
Comment (jimmyhogoboom) [inline packages/apollos-data-connector-postgres/src/people/dataSource.js:95]: Great idea!

File: packages/apollos-data-connector-postgres/src/postgres/index.js [hunk 2]
```diff
@@ -81,6 +81,18 @@ class PostgresDataSource {
...
 }
 
+const UUID_V4_REGEXP = new RegExp(
+  /^[0-9A-F]{8}-[0-9A-F]{4}-4[0-9A-F]{3}-[89AB][0-9A-F]{3}-[0-9A-F]{12}$/i
+);
+
+export const assertUuid = (uuid, caller = '') => {
+  if (uuid && !UUID_V4_REGEXP.test(uuid)) {
+    throw new Error(
+      `ID ${uuid} is not a valid UUID. You are probally passing a Rock (or other) id to ${caller} when it expects a Postgres UUID.`
```
Comment (jimmyhogoboom) [inline packages/apollos-data-connector-postgres/src/postgres/index.js:91]: Small typo
```suggestion
      `ID ${uuid} is not a valid UUID. You are probably passing a Rock (or other) id to ${caller} when it expects a Postgres UUID.`
```

File: packages/apollos-data-connector-postgres/src/follows/dataSource.js [hunk 3]
```diff
@@ -95,6 +98,55 @@ class Follow extends PostgresDataSource {
...
+    // select people.*, follows.state
+    // as follow_id from people
+    // left outer join follows on (follows."requestPersonId" = 'current user id' and follows."followedPersonId" = people.id)
+    // where people.email in ('email list') and (follows.state != 'ACCEPTED' or follows.state is null);
+
+    return this.sequelize.models.people.findAll({
+      where: {
+        email: { [Op.in]: suggestedEmails },
+        id: { [Op.ne]: id },
+        '$followingRequests.state$': { [Op.or]: [null, 'REQUESTED'] },
```
Comment (jimmyhogoboom) [inline packages/apollos-data-connector-postgres/src/follows/dataSource.js:138]: Oh nice! I'm gonna steal this for my PR.

**Review by jimmyhogoboom — APPROVED** [review]
Looks good!

**Comment by vinnyjth** [comment]
Made some changes, ready for re-review :D

**Comment by jimmyhogoboom** [comment]
@vinnyjth What changed?

**Comment by vinnyjth** [comment]
@jimmyhogoboom I hit the accept button on the typo, accept you beat me to it based on the commit log.

**Comment by jimmyhogoboom** [comment]
Ah weird. I was wondering how that would work.

**Review by jimmyhogoboom — APPROVED** [review]

**Pull request merged**